Refactors DNA scanner & adds new traitor final obj with it

JohnFulpWillard opened the pull request:
## About The Pull Request

I'm adding a new Traitor Final objective where you basically get a DNA scanner, that can only scan animals, but can mutate you into a space dragon when you inject yourself with Space Carp DNA.

![image](https://user-images.githubusercontent.com/53777086/160239638-614eb9f8-6e36-4af8-963c-690871063dbd.png)
![image](https://user-images.githubusercontent.com/53777086/160239717-4e352a52-047d-4d05-96b4-02bfc67ee1c1.png)
![image](https://user-images.githubusercontent.com/53777086/160239720-1f6afe6d-32b9-4a11-8bdc-9cc5134881f2.png)

A lot of the code here was copy & pasted from Romerol objective, so it's pretty easy to understand I hope.

## Why It's Good For The Game

There's been negative opinions about both Romerol and Battlecruiser, one of these criticisms is that they are the only two there are. I'd like to add more so there's more variety to it.

I also like this as it adds a way for Space Dragons to be player-spawned, I've already shown my distaste for midround ghost antagonists, as I feel they reward people for doing nothing, with a murderbone antag. Not that this solves any of my problems, but I can rest somewhat happier knowing there's a way for non-ghosts to get an antag like Space dragon. (It's why I really loved Blob infection's addition)

## Changelog

:cl:
fix: DNA Vault's DNA scanners no longer accept Plasmamen, Skeleton, and Golems as DNA.
expansion: A new Traitor Final objective has been added: Harvest the DNA of a Space Carp, and use it to turn yourself into a Space Dragon!
/:cl:

JohnFulpWillard committed:
Refactors DNA scanner & adds new traitor final obj

JohnFulpWillard committed:
makes the objective actually roll, also deletes probe

deletes probe after use so it isn't re-used in case they somehow drop it??
Them turning into a dragon deletes all their equipment though so I doubt it ever would've been a problem, I'd rather be safe then sorry.

Kylerace commented on code/game/objects/items/dna_probe.dm:
```diff
@@ -0,0 +1,108 @@
+		if(hydro_tray.plant_status != HYDROTRAY_PLANT_HARVESTABLE) // So it's bit harder.
+			to_chat(user, span_alert("Plant needs to be ready to harvest to perform full data scan.")) //Because space dna is actually magic
+			return
+		stored_dna_plants[hydro_tray.myseed.type] = 1
```
is this supposed to be the number one or a TRUE? because it seems like a TRUE

Kylerace reviewed (approved):
code looks good but ill defer to watermelons design approval

JohnFulpWillard committed:
renames human dna to human, 1 to TRUE

JohnFulpWillard committed:
moves rift making to the antag datum

Makes Traitor space dragons permanently buffed, but can't make rifts, as thye don't have the space dragon antag datum. They solely KILL now.

JohnFulpWillard committed:
Merge branch 'master' into space-dragon

JohnFulpWillard committed:
fixes minor screw-up i did

JohnFulpWillard committed:
Merge branch 'space-dragon' of https://github.com/JohnFulpWillard/tgstation into space-dragon

JohnFulpWillard committed:
remove the power on antag removal too

JohnFulpWillard committed:
renames rift ability to be more clear
